server/index.js

```diff
@@ -170,18 +188,42 @@ app.get('/server/questions', questionAuthoringAuth, function(request, response){
 
 
 // Write audio responses
-app.post('/teachermoments/wav', AudioEndpoints.post(s3));
+app.post('/teachermoments/wav', AudioEndpoints.post(config.s3));
 
 
 // Related to the read path for reviewing responses, and for fetching audio files
-app.post('/server/reviews/create', ReviewLoginEndpoint.createReview({queryDatabase, mailgunEnv}));
```

Comment:
ahh yes, I talked a bit about this with relation to the mailgunEnv and s3 variables. After moving those within the config, these calls broke when I used config.mailgunEnv instead of mailgunEnv. I don't really understand what restrictions the curly brackets enforce but I noticed that by removing the curly brackets, things worked fine again. I'm still not sure why this happens so if we feel strongly about having the curly brackets, I can look into this more?

Reply:
Got it.  So the curly brackets aren't doing anything magical, they're either creating an object, or pulling values out of an object.

```
ReviewLoginEndpoint.createReview({queryDatabase, mailgunEnv})

// is the same as:
const object = {
  queryDatabase: queryDatabase,
  mailgunEnv: mailgunEnv
};
ReviewLoginEndpoint.createReview(object);
```

So that passes one argument to the function, an object with two keys.

In contrast:
```
ReviewLoginEndpoint.createReview(queryDatabase, mailgunEnv)
```
calls the function with two arguments.

I don't know the details of what you changed here, but what might have happened before is that the shorcut `{foo}` expands to `{foo: foo}`.  But you can't do `{config.foo}` since it doesn't make sense to convert that to a key and value as`{config.foo: config.foo}`.  In that case you can just say `{foo: config.foo}`.

Reply:
This makes sense now! I've fixed this back to using the curly brackets.